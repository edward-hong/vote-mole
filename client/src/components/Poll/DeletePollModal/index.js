import React from 'react'
import { Modal, ModalHeader, ModalBody, ModalFooter, Button } from 'reactstrap'

import BlackButton from '../../styled/BlackButton'

const DeletePollModal = ({ isOpen, toggle, deletePoll }) => (
	<Modal isOpen={isOpen} toggle={toggle}>
		<ModalHeader>Delete Poll</ModalHeader>
		<ModalBody>Are you sure you want to delete this poll?</ModalBody>
		<ModalFooter>
			<Button outline color="danger" onClick={deletePoll}>
				Delete
			</Button>{' '}
			<BlackButton outline onClick={toggle}>
				Cancel
			</BlackButton>
		</ModalFooter>
	</Modal>
)

export default DeletePollModal
