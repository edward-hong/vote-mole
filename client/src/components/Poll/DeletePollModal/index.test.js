import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import DeletePollModal from '.'

describe('DeletePollModal', () => {
	it('is null when `isOpen` is false', () => {
		const { container } = render(<DeletePollModal isOpen={false} />)

		expect(container.firstChild).toBeNull()
	})

	it('renders correctly when `isOpen` is true', () => {
		const { container } = render(<DeletePollModal isOpen={true} />)

		expect(container.firstChild).toMatchSnapshot()
	})

	it('calls delete poll handler when the `Delete` button is clicked', () => {
		const deletePoll = jest.fn()
		const { getByText } = render(
			<DeletePollModal isOpen={true} deletePoll={deletePoll} />,
		)

		fireEvent.click(getByText('Delete'))

		expect(deletePoll).toHaveBeenCalled()
	})

	it('calls toggle when the `Cancel` button is clicked', () => {
		const toggle = jest.fn()
		const { getByText } = render(
			<DeletePollModal isOpen={true} toggle={toggle} />,
		)

		fireEvent.click(getByText('Cancel'))

		expect(toggle).toHaveBeenCalled()
	})
})
