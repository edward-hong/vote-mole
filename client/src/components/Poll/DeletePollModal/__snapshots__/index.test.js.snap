// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletePollModal renders correctly when \`isOpen\` is true 1`] = `null`;
